app/controllers/result_filters/location_controller.rb

```diff
@@ -2,21 +2,70 @@ module ResultFilters
   class LocationController < ApplicationController
     include FilterParameters
 
+    before_action :create_view
+
     def new; end
 
     def create
-      if(!params[:l])
-        flash[:error] = "Please choose an option"
-        redirect_to location_path(params_without_unsupported_location_params)
+      build_params = strip(filter_params.clone)
+
+      case @view.selected_option
+      when @view.no_option
+        handle_no_option(build_params)
+      when @view.location_option
+        handle_location_option(build_params)
       else
-        redirect_to results_path(params_without_unsupported_location_params)
+        redirect_to results_path(build_params)
       end
     end
 
   private
 
-    def params_without_unsupported_location_params
-      filter_params.except(:lat, :lng, :rad, :query, :loc, :lq)
+    def handle_no_option(build_params)
+      flash[:error] = "Please choose an option"
```

Comment:
We've used `app/views/result_filters/_error_message.html.erb` in the other filters to make copy and formatting consistent.